src/ddmd/mtype.d

```diff
@@ -3075,6 +3075,13 @@ extern (C++) abstract class Type : RootObject
     {
         v.visit(this);
     }
+
+    final TypeFunction toTypeFunction()
+    {
+        if (ty != Tfunction)
+            assert(0);
```

Comment:
The `assert(0)` will trigger in release builds, this is deliberate. I don't need the message (which will be useless to anyone who doesn't know the compiler internals).